src/api/run.rkt

```diff
@@ -69,8 +69,38 @@
   (when (not (directory-exists? dir))
     (make-directory dir))
 
-  (define results (get-test-results tests #:threads threads #:seed seed #:profile true #:dir dir))
-  (define info (make-report-info (filter values results) #:note note #:seed seed))
+  (start-job-server threads)
+  (define-values (job-ids bench-names)
+    (for/lists
+     (l1 l2)
+     ([test tests])
+     (define command
+       (create-job 'improve test #:seed seed #:pcontext #f #:profile? #f #:timeline-disabled? #f))
+     (values (start-job command) (test-name test))))
+
+  (define results
+    (for/list ([id job-ids]
+               [bench-name bench-names]
+               [i (in-naturals 0)])
+      (define result (wait-for-job id))
+      (define report-path (bench-folder-path bench-name i))
+      (define report-directory (build-path dir report-path))
+      (unless (directory-exists? report-directory)
+        (make-directory report-directory))
+
+      (for ([page (all-pages result)])
+        (call-with-output-file
+         (build-path report-directory page)
+         #:exists 'replace
+         (λ (out)
+           (with-handlers ([exn:fail? (λ (e) ((page-error-handler result page out) e))])
+             (make-page page out result #t #f)))))
+
+      (define table-data (get-table-data-from-hash result report-path))
+      (print-test-result (+ i 1) (length job-ids) table-data)
+      table-data))
```

Comment:
This is good but let's move this to a helper function, call it `generate-bench-report` or something.